src/Combinatorics/Matroids/realization_space.jl

```diff
@@ -0,0 +1,989 @@
+            elseif col in Bs 
+                #determinant zero but set is a basis, i.e. M is not realizable
+                return MatroidRealizationSpace(F, char, q)
+            else
+                continue
+            end
+            
+        end
+        
+        if  col in Bs
+            push!( ineqs, col_det )
+        else
+            push!( eqs, col_det )
+        end
+        
+    end
+
+    if q != nothing
+        for i in 1:polyR.nvars
+            push!(eqs, polyR[i]^q-polyR[i])
+        end
+    end
+
+    def_ideal = ideal(polyR,eqs)
+    def_ideal = ideal(groebner_basis(def_ideal))
+    isone(def_ideal) && return MatroidRealizationSpace(F, char, q)
+
+    # Unclear if we should use this. Can be too slow in some cases.
+    #=
+    if !iszero(def_ideal)
+        for i in 1:length(ineqs)
+            ineqs[i] = Oscar.reduce(ineqs[i], gens(def_ideal))
+        end
+    end
+    (polyR(0) in ineqs) && return MatroidRealizationSpace(nothing, nothing, nothing, nothing, false, F, char, q)
+    =#
+
+    ineqs = gens_2_factors(ineqs)
+    #=
+    # Only saturate easy inequations as it's too slow otherwise.
+    for i in 1:length(ineqs)
+        if count(k -> k!=0, degrees(ineqs[i])) < 3 && sum(degrees(ineqs[i])) < 3
+            def_ideal = saturation(def_ideal, ideal([ineqs[i]]))
+        end
+    end
+    =#
+    isone(def_ideal) && return MatroidRealizationSpace(F, char, q)
+    
+    RS = MatroidRealizationSpace(def_ideal, ineqs, polyR, mat, F, char, q)
+
+    if simplify
+        RS = reduce_realization_space(RS)
+    end
+
+    if saturate
+        RS.defining_ideal = stepwise_saturation(def_ideal,ineqs)
+        isone(RS.defining_ideal) && return MatroidRealizationSpace(F, char, q)
+        RS.realizable = true
+    end
+
+    return RS
+end
+
+
+# A heuristic function that tries to find a sensible basis for the moduli space computation for which the defining ideal is not too complicated
+function find_good_basis_heuristically(M::Matroid)
+    bs = bases(M)
+    cs = circuits(M)
+    min_num_vars = length(cs)*rank(M)
+    min_basis = bs[1]
+    for bi in 1:length(bs)
+        current_num_vars = 0
+        for c in cs
+            for e in c
+                @inbounds if e in bs[bi]
+                    current_num_vars += 1
+                end
+            end
+        end
+        if current_num_vars < min_num_vars
+            min_num_vars = current_num_vars
+            min_basis = bs[bi]
+        end
+    end
+    return min_basis
+end
+
+
+# Return the factors of f, but not the exponents
+function poly_2_factors(f::RingElem)
+    return collect(keys(Dict(factor(f))))
+end
+
+# Return the unique factors of the elements of Sgen, again no exponents. 
+function gens_2_factors(Sgens::Vector{<:RingElem})
+    return unique!(vcat([poly_2_factors(f) for f in Sgens]...))
+end
+
+
+function stepwise_saturation(I::MPolyIdeal, Sgens::Vector{<:RingElem})
+    foreach(f -> I = saturation(I,ideal([f])), Sgens)
```

Comment:
Don't do that, that abuses `foreach` for something it wasn't meant to do at all. Either write a genuine `for` loop:
```suggestion
    for f in Sgens
        I = saturation(I, ideal([f]))
    end
```
Or use `reduce` resp. `mapreduce`... something like this (untested)
```suggestion
    return mapreduce(f->ideal([f]), saturation, Sgens; init=I)
```